modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java

```diff
@@ -970,6 +976,38 @@ private RenderedImage renderImage(final GridCoverage2DReader reader,
             }
         }
         
+        // Check if reader supports band selection, and rearrange raster channels order in
+        // symbolizer. Reader should have taken care o proper channel order, based on initial
+        // symbolizer channel definition
+        if (reader.getFormat().getReadParameters().getDescriptor().descriptors().
+        		contains(AbstractGridFormat.BANDS) && readParams!=null){
+	        // if bands are selected, alter the symbolizer to use bands in order 0,1,2,... since the
+	        // channel order defined by it previously is taken care of the reader
+            int[] bandIndices = null;
+        	for (int i = 0; i < readParams.length; i++) {
```

Comment:
Formatting issues here, there are tabs (rule, make sure new code abides to the formatting rules, but don't reformat pre-existing code). Most IDEs have facilities to only format selected portions of code, or reformat on save only what has been modified.